lib/rules/prefer-const.js

```diff
@@ -300,7 +426,18 @@ module.exports = {
 
         return {
             "Program:exit"() {
-                groupByDestructuring(variables, ignoreReadBeforeAssign).forEach(checkGroup);
+                const identifierMap = groupByDestructuring(variables, ignoreReadBeforeAssign);
+
+                verifyAllDestructuring(identifierMap, checkingMixedDestructuring, destructureCount, destructureIdentifier).forEach(checkGroup);
```

Comment:
This doesn't seem right to me -- `destructureCount` is getting reassigned on every `ExpressionStatement`, so only the final value is getting used in the `Program:exit` listener. It seems like it doesn't make sense for the previous values of `destructureCount` to get discarded.